### PR TITLE
feat(core): add a `resolve_value` method

### PR DESCRIPTION
**Pull request by caspervonb**
This adds a `resolve_value` function which will return wait for a value to resolve and return it.
If the value is a promise and it rejects, that is returned as the error.

### CODE REVIEW

File: core/runtime.rs [hunk 1]
```diff
@@ -648,6 +648,38 @@ impl JsRuntime {
...
   }
 
+  /// Waits for the given value to resolve while polling the event loop.
+  ///
+  pub async fn resolve_value(
+    &mut self,
+    global: v8::Global<v8::Value>,
+  ) -> Result<v8::Global<v8::Value>, AnyError> {
+    poll_fn(|cx| {
+      self.poll_event_loop(cx, false);
```
Comment (caspervonb) [inline core/runtime.rs:658]: Suppose we should bail if the main loop poll yields `Poll::Ready` as nothing can progress?

File: core/runtime.rs [hunk 2]
```diff
@@ -1629,6 +1661,42 @@ pub mod tests {
...
+      let value = result_global.get(scope);
+      assert_eq!(value.integer_value(scope).unwrap(), 4);
+    }
+
+    let value_global = runtime
+      .execute_script("a.js", "Promise.reject(new Error('fail'))")
+      .unwrap();
+    let err = runtime.resolve_value(value_global).await.unwrap_err();
+    assert_eq!(
+      "Uncaught Error: fail",
```
Comment (caspervonb) [inline core/runtime.rs:1695]: Preferably this wouldn't be treated as uncaught
```suggestion
      "Error: fail",
```

File: core/runtime.rs (hunk 1 above)
Comment (bartlomieju) [reply core/runtime.rs:658]: Yes

File: core/runtime.rs (hunk 2 above)
Comment (bartlomieju) [reply core/runtime.rs:1695]: I'm afraid this will require to add a lot of conditionals to handle it just for this helper function

**Comment by bartlomieju** [comment]
Could we combine this PR with https://github.com/denoland/deno/pull/11727 as it's directly required?

**Comment by bartlomieju** [comment]
Thanks, closing in favor of #11727

**Pull request closed without merging**